### PR TITLE
BH eu2018: MetaFortress native obfuscator

### PR DESCRIPTION
**Pull request by enovella**
Close issue https://github.com/rednaga/APKiD/issues/112

### CODE REVIEW

File: apkid/rules/elf/obfuscators.yara [hunk 1]
```diff
@@ -215,3 +215,18 @@ rule avdobfuscator : obfuscator
...
+{
+  meta:
+    description = "MetaFortress"
+    url         = "https://www.insidesecure.com/Products/Application-Protection/Software-Protection/Code-Protection"
+    example     = "326632f52eba45609f825ab6746037f2f2b47bfe66fd1aeebd835c8031f4fdb0"
+    author      = "Eduardo Novella"
+
+  strings:
+    $a = "MetaFortress" nocase
+    $b = "METAFORIC"
```
Comment (CalebFenton) [inline apkid/rules/elf/obfuscators.yara]: Can these rules be full dex strings? I.e. null byte, leb128 length, string bytes, null byte?

Comment (enovella) [reply apkid/rules/elf/obfuscators.yara]: Added this `MetaFortress: %s`

Comment (CalebFenton) [reply apkid/rules/elf/obfuscators.yara]: So these can't be full c strings?? Nothing to the left or the right of the string?

**Comment by enovella** [comment]
The `METAFORIC` string dont remember from which sample took it, so I don't know if there was something left or right now. Is this okay?

File: apkid/rules/elf/obfuscators.yara (hunk 1 above)
Comment (enovella) [reply apkid/rules/elf/obfuscators.yara]:
```sh
[19:23 edu@fuckyou metafortress] >  r2 libmfjava.so 
 -- ESIL ruined my life
[0x00002a20]> izz~+metafortress
16908 0x0016e05f 0x0016e05f  12  13 (.rodata) ascii MetaFortress
16915 0x0016e187 0x0016e187  16  17 (.rodata) ascii MetaFortress: %s
[0x00002a20]> x 32 @ 0x0016e187-4
- offset -   0 1  2 3  4 5  6 7  8 9  A B  C D  E F  0123456789ABCDEF
0x0016e183  2f3e 0a00 4d65 7461 466f 7274 7265 7373  />..MetaFortress
0x0016e193  3a20 2573 0025 733a 2025 730a 0044 4542  : %s.%s: %s..DEB
[0x00002a20]> izz~+metafo
16908 0x0016e05f 0x0016e05f  12  13 (.rodata) ascii MetaFortress
16915 0x0016e187 0x0016e187  16  17 (.rodata) ascii MetaFortress: %s
[0x00002a20]> x 32 @ 0x0016e05f-4
- offset -   0 1  2 3  4 5  6 7  8 9  A B  C D  E F  0123456789ABCDEF
0x0016e05b  2e73 6f00 4d65 7461 466f 7274 7265 7373  .so.MetaFortress
0x0016e06b  0043 616e 6e6f 7420 646c 6f70 656e 206c  .Cannot dlopen l

```

File: apkid/rules/elf/obfuscators.yara [hunk 2]
```diff
@@ -215,3 +215,19 @@ rule avdobfuscator : obfuscator
...
+  meta:
+    description = "MetaFortress"
+    url         = "https://www.insidesecure.com/Products/Application-Protection/Software-Protection/Code-Protection"
+    sample      = "326632f52eba45609f825ab6746037f2f2b47bfe66fd1aeebd835c8031f4fdb0"
+    author      = "Eduardo Novella"
+
+  strings:
+    $a = { 00 4d65 7461 466f 7274 7265 7373 3a20 2573 0025 733a 2025 730a 00 } // MetaFortress %s.%s: %s
+    $b = { 00 4d65 7461 466f 7274 7265 7373 00 } // MetaFortress
+    $c = "METAFORIC"
```
Comment (CalebFenton) [inline apkid/rules/elf/obfuscators.yara]: I'm seeing a general trend on over reliance on strings, especially naked strings. The main problem with these rules isn't that they may FP -- though it's possible -- it's that _other_, much better rules are possible. I don't think it's worth it to add _relatively_ weak rules when a little extra effort could make something much better.

Comment (enovella) [reply apkid/rules/elf/obfuscators.yara]: Yup agree. Let me find new samples to tweak a bit better the rule.

Comment (strazzere) [reply apkid/rules/elf/obfuscators.yara]: Any new samples used to revise this? Or is it all set for next review?

Comment (enovella) [reply apkid/rules/elf/obfuscators.yara]: yeah! ETA soon

**Pull request closed without merging**